pkg/asset/machines/azure/machines.go

```diff
@@ -86,6 +86,20 @@ func provider(platform *azure.Platform, mpool *azure.MachinePool, osImage string
 
 	rg := platform.ClusterResourceGroupName(clusterID)
 
+	var image machineapi.Image
+	if mpool.OSImage.Publisher != "" {
+		image.Publisher = mpool.OSImage.Publisher
+		image.Offer = mpool.OSImage.Offer
+		image.SKU = mpool.OSImage.SKU
+		image.Version = mpool.OSImage.Version
```

Comment:
You also need to set `image.Type = machineapi.AzureImageTypeMarketplaceWithPlan`

Reply:
Resolve with aaba77d4a...d930460a7.